coral-hive/src/test/java/com/linkedin/coral/hive/hive2rel/HiveTableTest.java

```diff
@@ -63,6 +63,34 @@ public void testTable() throws Exception {
     RelDataTypeField colC = rowType.getField("c", false, false);
     assertEquals(colC.getType().getSqlTypeName(), SqlTypeName.ARRAY);
     assertEquals(colC.getType().getComponentType().getSqlTypeName(), SqlTypeName.DOUBLE);
+
+    // test handling of union
+    Table unionTable = getTable("default", "union_table");
+    // union_table:(foo uniontype<int, double, array<string>, struct<a:int,b:string>>)
+    // expected outcome schema: struct<tag:int, field0:int, field1:double, field2:array<string>, field3:struct<a:int,b:string>>
+    rowType = unionTable.getRowType(typeFactory);
+    assertNotNull(rowType);
+    assertTrue(rowType.isStruct());
+    // top-level: it is just a struct
+    assertEquals(rowType.getFieldCount(), 1);
```

Comment:
Can we refactor the assertions so instead of checking on things like field count of certain field names, they assert the whole expected schema?

Reply:
I figured the easiest way to do such assertion is to create a table with equivalent exploded union schema. If we would like to have a utility method to achieve "given a schema described in certain way and return the equivalent RelDataType", it will be pretty much the same thing as the visitor in Hive type to RelDataType conversion which I felt redundant.

Reply:
Does `RelDataType#toString()` or `RelDataType#getFullTypeString()` help?

Reply:
+1 to use `rowType.toString()` here so that we don't need to create another table `exploded_union` for test.